localstack/aws/spec.py

```diff
@@ -70,6 +70,10 @@ def load_service(service: ServiceName, version: str = None, model_type="service-
     For example: load_service("sqs", "2012-11-05")
     """
     service_description = loader.load_service_model(service, model_type, version)
+    # if the service name is sqs-json, we just loaded our internalized SQS query spec,
+    # the service name needs to be set to standard sqs
+    if service == "sqs-json":
+        service = "sqs"
     return ServiceModel(service_description, service)
```

Comment:
It would be really good if we could get `protocol: str` as an optional parameter into our `load_service` implementation instead of relying on the fake service names. i understand that we currently need them under the hood to know which spec file to load, basically our api should understand protocols as first-class citizen.

Reply:
This is a good point. The new solution does not use the `sqs-json` service anymore in the code.
However, using a custom service name has a big benefits: This way it integrates really nicely with `botocore` and our internal AWS client factory. We can just use `aws_client.sqs_json` to load a client which reads the spec we ship in the localstack module. This can be really helpful in the tests, and might also come in handy for other use cases. So we would like to have two ways to load a service with a specific protocol:
- Add the suffix to the service name (f.e. `sqs-json`), which enables the use case mentioned above.
- Add the protocol to the service loading function, to have a clean and clear way to define a specific protocol of a service to load.

I also tried to avoid patching `botocore`, but since the `spec#load_service` is the only function using the custom loader, which is why this custom logic (implementing the protocol-specific loading) is contained in that function.